src/sage/schemes/elliptic_curves/ell_finite_field.py

```diff
@@ -610,6 +610,24 @@ def cardinality(self, algorithm=None, extension_degree=1):
 
     order = cardinality  # alias
 
+    @cached_method
+    def multiplication_by_p_isogeny(self):
+        r"""
+        Return the multiplication-by-`p` isogeny.
+
+        EXAMPLES::
+
+            sage: p = 23
+            sage: K.<a> = GF(p^3)
+            sage: E = EllipticCurve(K, [K.random_element(), K.random_element()])
```

Comment:
This will throw an `ArithmeticError` whenever the discriminant of the elliptic curve happens to be zero; same issue in some more places below. (Sorry for not catching this earlier...)